redisinsight/api/src/modules/redis/redis.client.factory.ts

```diff
@@ -81,12 +81,16 @@ export class RedisClientFactory implements OnModuleInit {
     options: IRedisConnectionOptions = {},
   ): Promise<RedisClient> {
     const opts = RedisClientFactory.prepareConnectionOptions(options);
+    let connectionStrategy = await this.getConnectionStrategy(opts.clientLib);
+
+    if (clientMetadata.context === ClientContext.Profiler) {
+      connectionStrategy = this.ioredisConnectionStrategy;
+    }
```

Comment:
I think that we shouldn't put such logic to core functionality. I prefer to send `strategy` enum from profiler as an argument to force assign required one.